src/Components/Analyzers/src/ComponentFacts.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Components.Analyzers
+{
+    internal static class ComponentFacts
```

Comment:
This is part of the design that I've settled on for most analyzers we're building in ASP.NET Core.

Basically:
1. Big chunky analyzers that implement many features allows for efficient data sharing
2. Checking for symbols up front, and using symbols rather than syntax is productive and simple
3. Pulling logic into static methods based on symbols keeps code free of clutter and duplication